diff --git a/dist/node/chunks/node.js b/dist/node/chunks/node.js
index 156ecf04fdd966d1d61b633503309f72ac86e3a9..18b13e5fae1f82f89f6bc48f63d6735ba0bbf62c 100644
--- a/dist/node/chunks/node.js
+++ b/dist/node/chunks/node.js
@@ -2447,6 +2447,7 @@ function optimizeSafeRealPathSync() {
 	});
 }
 function ensureWatchedFile(watcher, file, root) {
+	if (watcher.closed) return;
 	if (file && !file.startsWith(withTrailingSlash(root)) && !file.includes("\0") && fs.existsSync(file)) watcher.add(path.resolve(file));
 }
 function joinSrcset(ret) {
@@ -35165,6 +35166,8 @@ function createDepsOptimizer(environment) {
 			depsOptimizer.scanProcessing,
 			optimizationResult?.cancel()
 		]);
+		depOptimizationProcessing.resolve();
+		resolveEnqueuedProcessingPromises();
 	}
 	let initState = "idle";
 	async function init() {
